data-prepper-plugins/saas-source-plugins/jira-source/src/test/java/org/opensearch/dataprepper/plugins/source/jira/JiraServiceTest.java

```diff
@@ -1,39 +1,260 @@
+        SearchResults results = jiraService.getAllIssues(jql, 0, jiraSourceConfig);
+        assertNotNull(results);
+    }
+
+    @Test
+    public void testGetAllIssuesInvalidAuthType() throws JsonProcessingException {
+        issueType.add("Task");
+        jiraSourceConfig = createJiraConfiguration("Invalid Auth Type", issueType);
+        jiraService = new JiraService(restTemplate, jiraSourceConfig, authConfig);
+        assertThrows(BadRequestException.class, () -> {
+            jiraService.getAllIssues(jql, 0, jiraSourceConfig);
+        });
+    }
+
+    private JiraSourceConfig createJiraConfiguration(String auth_type, List<String> issueType) throws JsonProcessingException {
+        ObjectMapper objectMapper = new ObjectMapper();
+        Map<String, String> connectorCredentialsMap = new HashMap<>();
+        connectorCredentialsMap.put("auth_type", auth_type);
+
+        Map<String, Object> jiraSourceConfigMap = new HashMap<>();
+        jiraSourceConfigMap.put("account_url", ACCESSIBLE_RESOURCES);
+        jiraSourceConfigMap.put("connector_credentials", connectorCredentialsMap);
+        jiraSourceConfigMap.put("issue_type", issueType);
+
+        String jiraSourceConfigJsonString = objectMapper.writeValueAsString(jiraSourceConfigMap);
+        return objectMapper.readValue(jiraSourceConfigJsonString, JiraSourceConfig.class);
+    }
+
+    private IssueBean createIssueBean(boolean nullFields) {
+        IssueBean issue1 = new IssueBean();
+        issue1.setId("123");
+        issue1.setKey("issue_1_key");
+        issue1.setSelf("https://example.com/rest/api/2/issue/123");
+        issue1.setExpand("operations,versionedRepresentations,editmeta");
+
+        Map<String, Object> fieldMap = new LinkedTreeMap<>();
+        if (!nullFields) {
+            fieldMap.put(CREATED, Instant.now());
+            fieldMap.put(UPDATED, Instant.now());
+        } else {
+            fieldMap.put(CREATED, 0);
+            fieldMap.put(UPDATED, 0);
+        }
+
+        Map<String, Object> issueTypeMap = new LinkedTreeMap<>();
+        issueTypeMap.put("name", "Task");
+        issueTypeMap.put("self", "https://example.com/rest/api/2/issuetype/1");
+        issueTypeMap.put("id", "1");
+        fieldMap.put("issuetype", issueTypeMap);
+
+        Map<String, Object> projectMap = new LinkedTreeMap<>();
+        if (!nullFields) {
+            projectMap.put("name", "project name test");
+            projectMap.put(KEY, "TEST");
+        }
+        fieldMap.put("project", projectMap);
+
+        Map<String, Object> priorityMap = new LinkedTreeMap<>();
+        priorityMap.put("name", "Medium");
+        fieldMap.put("priority", priorityMap);
+
+        Map<String, Object> statusMap = new LinkedTreeMap<>();
+        statusMap.put("name", "In Progress");
+        fieldMap.put("status", statusMap);
+
+        issue1.setFields(fieldMap);
+
+        return issue1;
+    }
+
+    private void waitForFutures() throws InterruptedException, TimeoutException {
```

Comment:
make this method, take futureList as an argument from each test

Reply:
ok